File: 1221-split-a-string-in-balanced-strings/1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        n = 0
        ans = 0
        for i in range(len(s)):
            if s[i] == 'L':
                n += 1
            else:
                n -= 1
            if n == 0:
                ans += 1

        return ans

File: 1221-split-a-string-in-balanced-strings/1221.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int count = 0;
        int i = 0;
        while (i < s.size()) {
            int balance = 0;
            for (int j = i; j < s.size(); ++j) {
                if (s[j] == 'R') {
                    ++balance;
                } else {
                    --balance;
                }
                if (balance == 0) {
                    ++count;
                    i = j + 1;
                }
            }
        }
        return count;
    }
};
